core/frontend/src/Viewport.ts

```diff
@@ -1,3331 +1,3331 @@
-       * in order to ensure browsers allow the underlying WebGL canvas to update. If a decorator is present, the 2d canvas
-       * is cleared during the frame render process by virtue of updating the decorator. For the non-decorator case, and for
-       * iOS, we must make sure we still clear the 2d canvas, done here. iOS appears to need this clear even when decorators
-       * clear the canvas later in the frame render process.
-       */
-      _clear2dCanvas(this.canvas);
-    }
-
-    this.target.updateViewRect();
-    this.invalidateRenderPlan();
-  }
-}
-
-function _clear2dCanvas(canvas: HTMLCanvasElement) {
-  const ctx = canvas.getContext("2d", { alpha: true })!;
-  ctx.save();
-  ctx.setTransform(1, 0, 0, 1, 0, 0); // revert any previous devicePixelRatio scale for clearRect() call below.
-  ctx.clearRect(0, 0, canvas.width, canvas.height);
-  ctx.restore();
-}
-
-/** Options supplied when creating an [[OffScreenViewport]].
- * @see [[OffScreenViewport.create]].
- * @public
- */
-export interface OffScreenViewportOptions {
-  /** The view to be drawn in the viewport. */
-  view: ViewState;
-  /** The dimensions of the viewport. */
-  viewRect: ViewRect;
-  /** If true, the viewport's aspect ratio will remain fixed. */
-  lockAspectRatio?: boolean;
-}
-
-/** A viewport that draws to an offscreen buffer instead of to the screen. An offscreen viewport is never added to the [[ViewManager]], therefore does not participate in
- * the render loop. Its dimensions are specified directly instead of being derived from an HTMLCanvasElement, and its renderFrame function must be manually invoked.
- * Offscreen viewports can be useful for, e.g., producing an image from the contents of a view (see [[Viewport.readImage]] and [[Viewport.readImageToCanvas]])
- * without drawing to the screen.
- * @public
- */
-export class OffScreenViewport extends Viewport {
-  protected _isAspectRatioLocked = false;
-
-  public static create(options: OffScreenViewportOptions): OffScreenViewport {
-    return this.createViewport(options.view, IModelApp.renderSystem.createOffscreenTarget(options.viewRect), options.lockAspectRatio);
-  }
-
-  /** @internal because RenderTarget is internal */
-  public static createViewport(view: ViewState, target: RenderTarget, lockAspectRatio = false): OffScreenViewport {
-    const vp = new this(target);
-    vp._isAspectRatioLocked = lockAspectRatio;
-    vp.changeView(view);
-    vp._decorationsValid = true;
-    return vp;
-  }
-
-  /** @internal */
-  public override get isAspectRatioLocked(): boolean {
-    return this._isAspectRatioLocked;
-  }
-
-  /** @internal */
-  public override get viewRect(): ViewRect {
-    return this.target.viewRect;
-  }
-
-  /** Change the dimensions of the viewport. */
-  public setRect(rect: ViewRect): void {
-    this.target.setViewRect(rect, false);
-    this.changeView(this.view);
-  }
-}
+/*---------------------------------------------------------------------------------------------
```

Comment:
Was this file changed due to line endings?

Reply:
gah yes, let me see if I can revert it.

Reply:
okay, should be fixed now.